crates/core/src/types/token.rs

```diff
@@ -1009,8 +1009,9 @@ pub struct MaspParams {
     pub kd_gain_nom: Dec,
     /// Shielded Pool nominal proportional gain for the given token
     pub kp_gain_nom: Dec,
-    /// Locked ratio for the given token
-    pub locked_ratio_target: Dec,
+    /// Target amount for the given token that is locked in the shielded pool
+    /// TODO: should this be a Uint or DenominatedAmount???
```

Comment:
It should be a uint, I think. Whatever the max supply of a token is (probably u256?)

Reply:
Ah but we have the weird MASP denom splitting thing, better think carefully about this

Reply:
`Uint` (or `Amount` in the worst case) should be fine here. The inflation computations are done at a high level, and the low-level MASP denom splitting is taken care of by the MASP conversion logic.

Reply:
Yeah I think `Uint` is best, as it is clearly readable and digestible by humans, whereas `Amount` will not be in this scenario.